build/xtask/src/sizes.rs

```diff
@@ -136,13 +136,25 @@ fn build_memory_map<'a>(
         .tasks
         .iter()
         .map(|(name, task)| {
-            (name.as_str(), &task.max_sizes, &allocs.tasks[name])
+            (
+                name.as_str(),
+                task.max_sizes.clone(),
+                allocs.tasks[name].clone(),
+            )
         })
         .chain(std::iter::once((
             "kernel",
-            &toml.kernel.requires,
-            &allocs.kernel,
+            toml.kernel.requires.clone(),
+            allocs.kernel.clone(),
         )))
+        .chain(allocs.caboose.iter().map(|(region, size)| {
+            let mut alloc = BTreeMap::new();
+            alloc.insert(region.clone(), size.clone());
+            let mut requires = IndexMap::new();
+            requires
+                .insert(region.clone(), toml.caboose.as_ref().unwrap().size);
+            ("-caboose-", requires, alloc)
```

Comment:
Other references to this region just call it `"caboose"` - does it matter that this one is `"-caboose-"`?

Reply:
This is for pretty-printing, so it shows up in the table as a special region:
```
flash:
      ADDRESS  |    PROGRAM     |  USED |  SIZE | LIMIT
    0x08000000 | kernel         | 18048 | 18048 | (fixed)
    0x08004680 | idle           |    68 |   128 | 128
    0x08004700 | -padding-      |    -- |  2304 |
    0x08005000 | jefe           |  3424 |  4096 | 4096
    0x08006000 | caboose_reader |  7424 |  8192 | 8192
    0x08008000 | hiffy          |  6568 |  8192 | 8192
    0x0800a000 | -caboose-      |   256 |   256 | 256
    0x0800a100 | --end--        |       |       |
```